src/clean-core/string_view.hh

```diff
@@ -131,6 +152,27 @@ public:
         return -1;
     }
 
+    /// returns the index of the start of the last occurrence of the character sequence (or -1 if not found)
+    constexpr int64 last_index_of(string_view s) const
+    {
+        if (s.size() > _size || s.empty())
```

Comment:
should non-empty be a contract instead?

Reply:
Yeah probably. At least I can't think of a use-case where searching for an empty string is not an error, or at least something that can't be caught on the user side.